phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -1081,7 +1083,11 @@ private RegionScanner rebuildIndices(final RegionScanner innerScanner, final Reg
             rawScan.setRaw(true);
             rawScan.setMaxVersions();
             rawScan.getFamilyMap().clear();
-            rawScan.setFilter(null);
+            if (scan.getFilter() instanceof FirstKeyOnlyFilter) {
+                rawScan.setFilter(null);
+            } else if (scan.getFilter() != null) {
+                rawScan.setFilter(new AllVersionsIndexRebuildFilter(scan.getFilter()));
+            }
```

Comment:
Do we need to do any special filter logic down at line 1099 in the else block if the Scan was raw in the first place?

Reply:
AFAIK we get raw scan here in case of old design and partial rebuild (Correct me if i'm wrong here @kadirozde ). I didn't want to mess with the old design and hence only made the changes for new.

Reply:
Yes, we get raw scan only for the old design partial rebuilds (i.e., auto-rebuilds).